Layout/LeadingCommentSpace: allow splitting long inline RBS comment signatures

Morriar opened the pull request:
At Shopify we use [RBS inline comments with Sorbet](https://sorbet.org/docs/rbs-support).

We added a syntax to allow splitting long signatures accross multiple lines like this:

```rb
#: (
#|   Integer,
#|   String
#| ) -> void
def foo; end
```

In combination with the syntax highlighting we added to the Ruby LSP, this makes it easier to read:

![image](https://github.com/user-attachments/assets/55bdf4d8-68b1-4167-9493-6698acccaa21)

The issue is that Rubocop's `Layout/LeadingCommentSpace` rule will try to add a space before the `|` even with `AllowRBSInlineAnnotation` enabled.

This PR proposes adding `#|` to the list of allowed characters for RBS inline comments.

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

koic commented:
It would be good to add this annotation style to example code in the documentation. Can you add it to example code?
https://github.com/rubocop/rubocop/blob/v1.75.4/lib/rubocop/cop/layout/leading_comment_space.rb#L52-L68

Also, the spec for the change log entry file format is failing. So, `bundle exec rspec spec/project_spec.rb` must pass.

Morriar commented:
> It would be good to add this annotation style to example code in the documentation. Can you add it to example code? https://github.com/rubocop/rubocop/blob/v1.75.4/lib/rubocop/cop/layout/leading_comment_space.rb#L52-L68

Good call, I added it 👍 

> Also, the spec for the change log entry file format is failing. So, `bundle exec rspec spec/project_spec.rb` must pass.

I corrected the log message, the passes now.

Thanks!

koic commented:
Thanks!
